oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/web/DefaultOAuth2AuthorizationRequestResolver.java

```diff
@@ -201,6 +207,28 @@ private static String expandRedirectUri(HttpServletRequest request, ClientRegist
 				.toUriString();
 	}
 
+	/**
+	 * Creates nonce and its hash for use in OpenID Connect Authentication Requests
+	 *
+	 * @param attributes where {@link IdTokenClaimNames#NONCE} is stored for the token request
+	 * @param additionalParameters where hash of {@link IdTokenClaimNames#NONCE} is added to the authentication request
+	 *
+	 * @since 5.2
+	 * @see <a target="_blank" href="https://openid.net/specs/openid-connect-core-1_0.html#NonceNotes">15.5.2.  Nonce Implementation Notes</a>
+	 * @see <a target="_blank" href="https://openid.net/specs/openid-connect-core-1_0.html#IDTokenValidation">3.1.3.7.  ID Token Validation</a>
+	 */
+	private void addNonceParameters(Map<String, Object> attributes, Map<String, Object> additionalParameters) {
+		String nonce = this.stringKeyGenerator.generateKey();
+		attributes.put(IdTokenClaimNames.NONCE, nonce);
+
+		try {
+			String nonceHash = createHash(nonce);
+			additionalParameters.put(IdTokenClaimNames.NONCE, nonceHash);
+		} catch (NoSuchAlgorithmException e) {
+			//
```

Comment:
Looks like this [comment](https://github.com/spring-projects/spring-security/pull/7337#discussion_r322893893) was missed.

If the `nonceHash` could not be created than the `nonce` should not be added to the `attributes`. We should add a test for this as well.

Reply:
Corrected. I was following the PKCE approach suggested, and they add an attribute even if they're unable to create the hash...but they also add that same unencrypted value if hashing fails too. :/ At any rate, I've change to only add nonce attribute if we can also add the hash to additional parameters.